aptos-move/move-examples/token_objects/ambassador/client/ambassador.ts

```diff
@@ -0,0 +1,171 @@
+            input: process.stdin,
+            output: process.stdout
+        });
+
+        rl.question('Please press the Enter key to proceed ...\n', () => {
+            rl.close();
+            resolve();
+        });
+    });
+}
+
+class AmbassadorClient extends AptosClient {
+    constructor() {
+        super(NODE_URL);
+    }
+
+    async setAmbassadorLevel(creator: AptosAccount, token: HexString, new_ambassador_level: number | bigint): Promise<string> {
+        const rawTxn = await this.generateTransaction(creator.address(), {
+            function: `${creator.address()}::ambassador::set_ambassador_level`,
+            type_arguments: [],
+            arguments: [token.hex(), new_ambassador_level],
+        });
+
+        const bcsTxn = await this.signTransaction(creator, rawTxn);
+        const pendingTxn = await this.submitTransaction(bcsTxn);
+
+        return pendingTxn.hash;
+    }
+
+    async burn(creator: AptosAccount, token: HexString): Promise<string> {
+        const rawTxn = await this.generateTransaction(creator.address(), {
+            function: `${creator.address()}::ambassador::burn`,
+            type_arguments: [],
+            arguments: [token.hex()],
+        });
+
+        const bcsTxn = await this.signTransaction(creator, rawTxn);
+        const pendingTxn = await this.submitTransaction(bcsTxn);
+
+        return pendingTxn.hash;
+    }
+
+    async mintAmbassadorToken(creator: AptosAccount, description: string, name: string, uri: string, soul_bound_to: HexString): Promise<string> {
+        const rawTxn = await this.generateTransaction(creator.address(), {
+            function: `${creator.address()}::ambassador::mint_ambassador_token`,
+            type_arguments: [],
+            arguments: [description, name, uri, soul_bound_to.hex()],
+        });
+
+        const bcsTxn = await this.signTransaction(creator, rawTxn);
+        const pendingTxn = await this.submitTransaction(bcsTxn);
+
+        return pendingTxn.hash;
+    }
+}
+
+/** run our demo! */
+async function main(): Promise<void> {
+    const client = new AmbassadorClient();
+
+    // Set a test admin account
+    const privateKeyBytes_admin = Uint8Array.from(Buffer.from('f21423f436f7d44c2abd95b5a25323e81fc737040ab17ae8fe40dbf1b1de9e66', 'hex'));
+    const admin = new AptosAccount(privateKeyBytes_admin, '9bfdd4efe15f4d8aa145bef5f64588c7c391bcddaf34f9e977f59bd93b498f2a');
+    // Set a test user account
+    const userAddr = new HexString("4db1582c315ddd9f29db3dfcf0aa7f7467b1a4f2d1190bb93b8304cdc164490c");
+    // Set a test token name
+    const tokenName = "Aptos Ambassador #23";
+
+    console.log("\n=== Addresses ===");
+    console.log(`Admin: ${admin.address()}`);
+    console.log(`User: ${userAddr}`);
+
+    // Mint Ambassador Token
+    let txnHash = await client.mintAmbassadorToken(admin, 'Aptos Ambassador Token', tokenName, 'https://raw.githubusercontent.com/junkil-park/metadata/main/ambassador_1/', userAddr);
```

Comment:
Can we dump the metadata into this directory so we don't need to point to the link?

Reply:
Good point. Done.